pkg/minikube/machine/cache_images.go

```diff
@@ -0,0 +1,147 @@
+			if err := CacheImage(image, dst); err != nil {
+				return errors.Wrapf(err, "caching image %s", dst)
+			}
+			if err := LoadFromCache(d, dst); err != nil {
+				return errors.Wrapf(err, "loading image from cache: %s", dst)
+			}
+			return nil
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return errors.Wrap(err, "caching images")
+	}
+
+	return nil
+}
+
+// # ParseReference cannot have a : in the directory path
+func sanitizeCacheDir(image string) string {
+	return strings.Replace(image, ":", "_", -1)
+}
+
+func LoadFromCache(d drivers.Driver, src string) error {
+	glog.Infoln("Loading image from cache at ", src)
+	filename := filepath.Base(src)
+	dst := filepath.Join(tempLoadDir, filename)
+	f, err := assets.NewFileAsset(src, tempLoadDir, filename, "0777")
+	if err != nil {
+		return errors.Wrapf(err, "creating copyable file asset: %s", filename)
+	}
+
+	client, err := sshutil.NewSSHClient(d)
+	if err != nil {
+		return errors.Wrap(err, "getting ssh client")
+	}
+	if err := sshutil.TransferFile(f, client); err != nil {
+		return errors.Wrap(err, "transferring cached image")
+	}
+
+	s, err := client.NewSession()
+	if err != nil {
+		return errors.Wrap(err, "getting ssh session")
+	}
+
+	dockerLoadCmd := "docker load -i " + dst
+	if err := s.Run(dockerLoadCmd); err != nil {
+		return errors.Wrapf(err, "loading docker image: %s", dst)
+	}
+
+	return nil
+}
+
+func getSrcRef(image string) (types.ImageReference, error) {
+	srcRef, err := docker.ParseReference("//" + image)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing docker image src ref")
+	}
+	return srcRef, nil
+}
+
+func getDstRef(image, dst string) (types.ImageReference, error) {
+	dstRef, err := archive.ParseReference(dst + ":" + image)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing docker archive dst ref")
+	}
+	return dstRef, nil
+}
+
+func CacheImage(image, dst string) error {
+	glog.Infof("Attempting to cache image: %s at %s\n", image, dst)
+	if _, err := os.Stat(dst); err == nil {
+		return nil
+	}
+
+	if err := os.MkdirAll(filepath.Dir(dst), 0777); err != nil {
+		return errors.Wrapf(err, "making cache image directory: %s", dst)
+	}
+
+	srcRef, err := getSrcRef(image)
+	if err != nil {
+		return errors.Wrap(err, "creating docker image src ref")
+	}
+
+	dstRef, err := getDstRef(image, dst)
+	if err != nil {
+		return errors.Wrap(err, "creating docker archive dst ref")
+	}
+
+	policy, err := signature.DefaultPolicy(nil)
+	if err != nil {
+		return errors.Wrap(err, "getting default signature policy")
+	}
+
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return errors.Wrap(err, "getting policy context")
+	}
+
+	sourceCtx := &types.SystemContext{
+		// By default, the image library will try to look at /etc/docker/certs.d
+		// As a non-root user, this would result in a permissions error,
+		// so, we skip this step by just looking in the minikube home directory.
+		DockerCertPath: constants.MakeMiniPath("cache"),
```

Comment:
cc @dlorenc @cftorres this was my trick to get around the /etc/docker/certs.d error.